customer-managed/aws/terraform/iam_redpanda_agent.tf

```diff
@@ -337,7 +337,7 @@ data "aws_iam_policy_document" "redpanda_agent2" {
       aws_iam_policy.load_balancer_controller_policy["1"].arn,
       aws_iam_policy.load_balancer_controller_policy["2"].arn,
       # redpanda_agent1 and redpanda_agent2, cannot be referenced by object due to cycle
-      "arn:aws:iam::${local.aws_account_id}:policy/redpanda-agent-*-*",
+      "arn:aws:iam::${local.aws_account_id}:policy/${var.common_prefix}agent-*-*",
```

Comment:
This feels a bit unatural as the prefix needs to include a separator like `myprefix-` vs `myprefix`.  But I understand the counter-argument of "What if customer doesn't want a dash" (but there's a dash anyway for `-*-*`)

Reply:
I can take the dash out of the variable, but that seems unrelated to this change.

Reply:
https://github.com/redpanda-data/cloud-examples/pull/11